src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -0,0 +1,483 @@
+     */
+    public T setOtherApps(List<String> apps) {
+        return amend(MobileCapabilityType.OTHER_APPS, apps);
+    }
+
+    /**
+     * Get the list of apps to install before running a test.
+     *
+     * @return List of apps to install.
+     * @see MobileCapabilityType#OTHER_APPS
+     */
+    public List<String> getOtherApps() {
+        return (List<String>) getCapability(MobileCapabilityType.OTHER_APPS);
+    }
+
+    /**
+     * Set the version of the platform.
+     *
+     * @param version is the platform version.
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#PLATFORM_VERSION
+     */
+    public T setPlatformVersion(String version) {
+        return amend(MobileCapabilityType.PLATFORM_VERSION, version);
+    }
+
+    /**
+     * Get the version of the platform.
+     *
+     * @return String representing the platform version.
+     * @see MobileCapabilityType#PLATFORM_VERSION
+     */
+    public String getPlatformVersion() {
+        return (String) getCapability(MobileCapabilityType.PLATFORM_VERSION);
+    }
+
+    /**
+     * Set the app to print page source when a find operation fails.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#PRINT_PAGE_SOURCE_ON_FIND_FAILURE
+     */
+    public T setPrintPageSourceOnFindFailure() {
+        return setPrintPageSourceOnFindFailure(true);
+    }
+
+    /**
+     * Set whether the app to print page source when a find operation fails.
+     *
+     * @param bool is whether to print page source.
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#PRINT_PAGE_SOURCE_ON_FIND_FAILURE
+     */
+    public T setPrintPageSourceOnFindFailure(boolean bool) {
+        return amend(MobileCapabilityType.PRINT_PAGE_SOURCE_ON_FIND_FAILURE, bool);
+    }
+
+    /**
+     * Get whether the app to print page source when a find operation fails.
+     *
+     * @return true if app prints page source.
+     * @see MobileCapabilityType#PRINT_PAGE_SOURCE_ON_FIND_FAILURE
+     */
+    public boolean doesPrintPageSourceOnFindFailure() {
+        return (boolean) getCapability(MobileCapabilityType.PRINT_PAGE_SOURCE_ON_FIND_FAILURE);
+    }
+
+    /**
+     * Set the id of the device.
+     *
+     * @param id is the unique device identifier.
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#UDID
+     */
+    public T setUdid(String id) {
+        return amend(MobileCapabilityType.UDID, id);
+    }
+
+    /**
+     * Get the id of the device.
+     *
+     * @return String representing the unique device identifier.
+     * @see MobileCapabilityType#UDID
+     */
+    public String getUdid() {
+        return (String) getCapability(MobileCapabilityType.UDID);
+    }
+
+    @Override
+    public T merge(Capabilities extraCapabilities) {
+        super.merge(extraCapabilities);
+        return (T) this;
+    }
+
+    protected T amend(String optionName, Object value) {
```

Comment:
is there a possibility to set a custom capability name as string?

Reply:
Well, this is a private method. Users can either use `setCapability()` or we can handle specific converting custom capabilities to String keys in the public setter.